src/test/ui/async-await/issue-70594.stderr

```diff
@@ -25,6 +25,7 @@ LL |     [1; ().await];
    |         ^^^^^^^^ `()` is not a future
    |
    = help: the trait `Future` is not implemented for `()`
+   = note: required because of the requirements on the impl of `IntoFuture` for `()`
```

Comment:
I don't think this is very clear because it's not obvious to the uninitiated user where `IntoFuture` came from. We can improve it with a [`rustc_on_unimplemented`](https://rustc-dev-guide.rust-lang.org/diagnostics.html#rustc_on_unimplemented) attribute on `IntoFuture`. As an example..

> note: the type `()` must be a future or implement `IntoFuture` directly to be awaited